cartridges/int_extend/cartridge/scripts/extendAnalyticsHelpers.js

```diff
@@ -0,0 +1,290 @@
+            var item = productLineItems[j];
+            if (item.UUID === product.UUID) {
+                isProductExist = true;
+                if (product.quantityValue !== item.quantityValue) {
+
+                    if (item.custom.persistentUUID) {
+                        extendProduct = getExtendProduct(productLineItems, item);
+
+                        if (!extendProduct) {
+                            data.event = 'productUpdated'
+                            itemsData.push(getProductUpdatedData(item, data));
+                        }
+
+                    } else if (item.custom.parentLineItemUUID) {
+                        extendedProduct = getExtendedProduct(productLineItems, item);
+                        data.event = 'offerUpdated'
+                        var newData = getOfferUpdatedData(extendedProduct, item, data)
+
+                        if (product.quantityValue !== newData.warrantyQuantity) {
+                            itemsData.push(newData);
+                        }
+
+                    } else {
+                        data.event = 'productUpdated'
+                        itemsData.push(getProductUpdatedData(item, data));
+                    }
+                }
+            }
+        }
+
+        if (!isProductExist && +product.newQuantity === 0) {
+            data.productID = product.productID;
+            if (product.planId) {
+                data.planId = product.planId;
+                data.event = 'offerRemovedFromCart'
+            } else {
+                data.event = 'productRemovedFromCart'
+            }
+            itemsData.push(data);
+        }
+
+    }
+    var newData = {
+        array: itemsData
+    }
+
+    var strData = JSON.stringify(newData);
+
+    Transaction.wrap(function () {
+        request.session.custom.analyticsPayload = strData;
+    });
+}
+
+/**
+ * Set extend analytics data after delete product
+ * @param {Object} cart the cart model
+ * @param {Object} object product to delete
+ * @returns {void}
+ */
+function setDeleteProductPayload(cart, object) {
+    var Transaction = require('dw/system/Transaction');
+    var extendProduct, extendedProduct;
+    var itemsData = [];
+    var data = {};
+
+    if (object.custom.parentLineItemUUID) {
+        extendedProduct = getExtendedProduct(cart.object.productLineItems, object);
+        itemsData.push(getOfferRemovedFromCartData(extendedProduct, object));
+
+    } else if (object.custom.persistentUUID) {
+        extendProduct = getExtendProduct(cart.object.productLineItems, object);
+
+        if (!extendProduct) {
+            itemsData.push(getProductRemovedFromCartData(object));
+        }
+
+    } else {
+        itemsData.push(getProductRemovedFromCartData(object));
+    }
+
+    var newData = {
+        array: itemsData
+    }
+
+    var strData = JSON.stringify(newData);
+
+    Transaction.wrap(function () {
+        request.session.custom.analyticsPayload = strData;
```

Comment:
Can we update a viewData instead of the session?

Reply:
int_extend is cartridge for SG and It has no viewData functionality. Pdict was not used because at the end of the endpoint (where this function is called) a redirect to another endpoint is made and the result must be passed to it.